### DISCUSSION TITLE
GLIBC_2.18 not found (uws_linux_x64_72.node)

### DISCUSSION THREAD

**Discussion by codermapuche** [uNetworking/uWebSockets.js#218] [Q&A] [answered]
Hello when i try to use it, throw this error:

Error: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /home/uws/node_modules/uWebSockets.js/uws_linux_x64_72.node)

when i run:

> ldd --version
> ldd (GNU libc) 2.17

Is possible compile with 2.17? Is a dedicated server and can't change the entire operative system (CENTOS) or install 2.18 and run both.

How i can do it?

Thanks!

**Comment by ghost** [comment, marked as answer]
It looks like provided binaries depend on glibc 2.18 from 2013 - that seems alright. If you want to recompile for older versions you're on your own (this also applies to people wanting Alpine Linux binaries - same story).

**Comment by lostrepo** [comment]
@codermapuche
Building for CentOS.
1. Install `gcc` that supports `C++17` (you can see if required version changes in `build.c` file). I've built `uWebSocketsjs 0.16` with `gcc version 4.8.5 20150623 (Red Hat 4.8.5-39) (GCC)`
2. In `build.c` under `/* Linux */` change `clang` and `clang++` to `gcc` and `g++`
3. Grab `uWebSocketsjs`. If referenced repos with `uWebSockets` and `uSockets` didn't load - grab them too.
3. Follow general building from source based on `makefile`
4. From `uws/dist/` grab addons with js file and drop into your projects `node_modules/uWebsocketsjs/`
5. Enjoy fast http/ws server.

**Comment by dominicklee** [comment]
I have built a custom binary (and tutorial) that will solve your exact problem. Hope it helps!

Read it here: [https://github.com/dominicklee/uWebSockets.js-for-Centos7][1]

  [1]: https://github.com/dominicklee/uWebSockets.js-for-Centos7
